solutions/gold/kattis-quantumsuperposition.mdx

```diff
@@ -28,66 +28,238 @@ Once we know all possible path lengths for each universe, we can find all
 possible sums of lengths. Just loop through both universe's route lengths and
 add them together.
 
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bitset>
 #include <iostream>
 #include <queue>
 #include <vector>
 using namespace std;
 
+static const int MAX_N = 1000;
+static const int MAX_Q = 2000;
+
 int n[2], m[2];
-vector<int> g[2][1001];
-vector<int> back[2][1001];
-bitset<2001> dp[2][1001];
+vector<int> g[2][MAX_N + 1];
+vector<int> back[2][MAX_N + 1];
+bitset<MAX_Q + 1> dp[2][MAX_N + 1];
 
+/*
+ * Generates the topological sort, filling in the DP array in the process.
+ */
 void gen(int x) {
-	int in_degree[1001] = {};
+	int in_degree[MAX_N + 1] = {};
 	for (int i = 0; i < m[x]; i++) {
 		int a, b;
 		cin >> a >> b;
 		g[x][a].push_back(b);
 		back[x][b].push_back(a);
 		in_degree[b]++;
 	}
-	// finding length of routes of first universe
+	// Run topological sort.
 	queue<int> q;
 	for (int i = 0; i <= n[x]; i++) {
 		if (in_degree[i] == 0) { q.push(i); }
 	}
 	while (!q.empty()) {
 		int node = q.front();
 		q.pop();
-		// using dp while processing the nodes topologically
-		if (back[x][node].empty()) dp[x][node][0] = 1;
-		for (int before : back[x][node]) dp[x][node] |= dp[x][before] << 1;
 		for (int next : g[x][node]) {
 			in_degree[next]--;
-			if (in_degree[next] == 0) q.push(next);
+			if (in_degree[next] == 0) { q.push(next); }
 		}
+
+		// `node` can be visited in `i + 1` steps if `before` can in `i` steps.
+		if (back[x][node].empty()) { dp[x][node][0] = 1; }
+		for (int before : back[x][node]) { dp[x][node] |= dp[x][before] << 1; }
 	}
 }
 
 int main() {
 	ios_base::sync_with_stdio(0);
 	cin.tie(0);
```

Comment:
remove if runs w/o

Reply:
remove the entire lines? it doesnt work if i do

Reply:
he means the fast i/o fyii